### ISSUE TITLE
Fix annotation processing & update kotlin

### ISSUE THREAD

**Issue by Xerus2000** [sghpjuikit/player#55]
Updating the kotlin plugin to 1.2.30 in `project.gradle.kts` wants us to replace the deprecated `kapt` plugin with the newer `kotlin-kapt`, or it won't compile. The new plugin however seems to have some issues with the current code. This issue tracks the investigations in this problem.

**Comment by Xerus2000** [comment]
testing this here: https://github.com/sghpjuikit/player/tree/kapt

**Comment by Xerus2000** [comment]
Opened an issue here: https://youtrack.jetbrains.com/issue/KT-23114

**Comment by sghpjuikit** [comment]
It seems there are indeed some kapt issues. The issue above was caused by an enum class defined within another class. Moving the enum out fixed the issue. However another issue was encountered - unicode support. After painstaking removal of all unicode from the project another issue was revealed - kapt could not see some (or all?) of the dependencies and complained about unresolvable symbols.

Losing the nerve I decided to completely get rid of any annotation processing in the project. This makes it  possible to update to 1.2.30.

Some notes on the change:

- The only annotation processing going on was for the configuration framework to find all classes with static Config fields. This can easily be changed to manual registration of the classes. It is not ideal, but I'm working on more robust and simpler solution using Kotlin's companion objects and property delegates. This should also bring support for lazy loading of static Configs (not forcing developer to load the class when application starts) and particularly allow on-demand config loading (making it possible to request a not yet loaded config, type-safely at that).
- no annotation processing gives us several advantages, simpler project setup, less dependencies, faster & simpler build process, more gradle optimizations, more comprehensible code. In general there is less to reason about and less to go wrong.
